azure-functions-java-opentelemetry/src/main/java/com/microsoft/azure/functions/opentelemetry/OpenTelemetryInvocationMiddleware.java

```diff
@@ -0,0 +1,42 @@
+package com.microsoft.azure.functions.opentelemetry;
+
+import com.microsoft.azure.functions.internal.spi.middleware.Middleware;
+import com.microsoft.azure.functions.internal.spi.middleware.MiddlewareChain;
+import com.microsoft.azure.functions.internal.spi.middleware.MiddlewareContext;
+
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.SpanKind;
+import io.opentelemetry.api.trace.StatusCode;
+import io.opentelemetry.context.Scope;
+
+
+/**
+ * Middleware that starts a Span for every function invocation,
+ * using the trace context from the host if available.
+ */
+public class OpenTelemetryInvocationMiddleware implements Middleware {
+
+    @Override
+    public void invoke(MiddlewareContext context, MiddlewareChain chain) throws Exception {
+        String spanName = context.getFunctionName();
+        String tracerName = "azure.functions.worker";
+
+        FunctionsOpenTelemetry.setLogger(context.getLogger());
+        Span invocationSpan = FunctionsOpenTelemetry.startSpan(spanName, tracerName, context.getTraceContext(), SpanKind.INTERNAL);
```

Comment:
[Checking] If with every function invocation we with configure the OTEL? 

Also when you plan to end the span? 

If above is the case then it is adding a performance issue. We should see if we can have the span started at the start of the function and end when terminated. Otel Span start with every invocation is overhead.

Reply:
My apologies for not familiar with the codebase, how that is lazy loading is supported here? I am not seeing any code which suggesting that. Also, why this cannot be done during functions registration?

My concern is will it impact cold start?